medusa/clients/torrent/downloadstation.py

```diff
@@ -39,195 +36,135 @@ def __init__(self, host=None, username=None, password=None):
-        self._request(method='post', data=data)
-        return self._check_response()
+            # parameters "type' and "destination" must be coded as a json string so we add double quotes to those parameters
+        params = {  'api'        : 'SYNO.DownloadStation2.Task',
+                    'version'    : 2,
+                    'method'     : 'create',
+                    'create_list': 'false',
+                    'destination': f'"{app.TORRENT_PATH}"',
+                    'type'       : '"url"',
+                    'url'        : result.url}
+        return self._add_torrent(params)
 
     def _add_torrent_file(self, result):
-        # The API in the latest version of Download Station (3.8.16.-3566)
-        # is broken for downloading via a file, only uri's are working correct.
-        if result.url[:4].lower() in ['http', 'magn']:
-            return self._add_torrent_uri(result)
-
-        torrent_path = app.TORRENT_PATH
-
-        data = {
-            'api': 'SYNO.DownloadStation.Task',
-            'version': '1',
-            'method': 'create',
-            'session': 'DownloadStation',
-        }
-
-        if not self._check_destination():
-            return False
-
-        if torrent_path:
-            data['destination'] = torrent_path
-
-        files = {'file': ('{name}.torrent'.format(name=result.name), result.content)}
-
-        log.debug('Add torrent files with data: {0}', json.dumps(data))
-        self._request(method='post', data=data, files=files)
-        return self._check_response()
-
-    def _check_destination(self):
-        """Validate and set torrent destination."""
-        torrent_path = app.TORRENT_PATH
-
-        if not (self.auth or self._get_auth()):
+            # parameters "type" and "file" must be individually coded as a json string so we add double quotes to those parameters
+            # the "file" paramater (torrent in this case) must corrospondent with the key in the files parameter
+        params = {'api'        : 'SYNO.DownloadStation2.Task',
+                  'version'    : 2,
+                  'method'     : 'create',
+                  'create_list': 'false',
+                  'destination': f'"{app.TORRENT_PATH}"',
+                  'type'       : '"file"',
+                  'file'       : '["torrent"]'}
+        torrent_file = {'torrent': (f'{result.name}.torrent', result.content)}
+        return self._add_torrent(params, torrent_file)
+
+
+    def _add_torrent(self, params, torrent_file=None):
+        """Add a torrent to DownloadStation"""
+        self.url = urljoin(app.TORRENT_HOST, 'webapi/entry.cgi')
+        if not self._request(method='post', data=params, files=torrent_file):
             return False
-
-        if self.checked_destination and self.destination == torrent_path:
+        jdata = self.response.json()
+        if jdata['success']:
+            log.info(f"Torrent added as task {jdata['data']['task_id']} to {self.name}")
```

Comment:
we don't use template literals for logs. We use a modified paramaterized string replacement. Like how python's logging module can do it by default with it's string replacement: `log.debug('my message with %s', my_param)`. But our BraceAdapter uses a dict as the parameter.

For ex: 
```
log.debug('{name}: Response to {method} request is {response}', {
            'name': self.name,
            'method': method.upper(),
            'response': self.response.text[0:1024] + '...' if len(self.response.text) > 1027 else self.response.text
        })
```

Reply:
Sorry, I think I accidentele hidden this one. Can you take a look at it?
Using string templating like `%s`, `{}` or f"" in the logging module, is not best practice. Can you change it to BraceAdapter?